nautobot/extras/models/groups.py

```diff
@@ -431,3 +446,199 @@ def clean(self):
 
         # Validate `filter` dict
         self.clean_filter()
+
+    def generate_query_for_filter(self, filter_field, value):
```

Comment:
Docstrings and comments please! There's a lot of reasonably complex code below here that isn't necessarily intuitive or obvious.